controlplane/kubeadm/controllers/controller.go

```diff
@@ -384,34 +384,32 @@ func (r *KubeadmControlPlaneReconciler) reconcileDelete(ctx context.Context, clu
 	logger := r.Log.WithValues("namespace", kcp.Namespace, "kubeadmControlPlane", kcp.Name, "cluster", cluster.Name)
 	logger.Info("Reconcile KubeadmControlPlane deletion")
 
+	// Gets all machines, not just control plane machines.
+	allMachines, err := r.managementCluster.GetMachinesForCluster(ctx, util.ObjectKey(cluster))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	ownedMachines := allMachines.Filter(machinefilters.OwnedMachines(kcp))
+
 	// Ignore the health check results here as well as the errors, health check functions are to set health related conditions on Machines.
 	// Errors may be due to not being able to get workload cluster nodes.
 	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
 	if err != nil {
 		r.Log.V(2).Info("Cannot get remote client to workload cluster during delete reconciliation", "err", err.Error())
 	} else {
 		// Do a health check of the Control Plane components
-		_, err = workloadCluster.ControlPlaneIsHealthy(ctx)
-		if err != nil {
+		if err := workloadCluster.ControlPlaneIsHealthy(ctx, ownedMachines.UnsortedList()); err != nil {
```

Comment:
`ownedMachines` used here should be ControlPlaneMachines.

`	controlPlaneMachines, err := r.managementCluster.GetMachinesForCluster(ctx, util.ObjectKey(cluster), machinefilters.ControlPlaneMachines(cluster.Name))
`

Reply:
I'm not sure we should use a different filter here.

In reconcile (normal) we are using ownedMachine as a source for building the controlPlane object/running health checks, so IMO it makes sense here as well.

Also, in the entire reconcile delete, the assumption is that all the machines owned by KCP are the only control plane machine existing in the cluster, and if there are reason to change this, we should change this in the entire func - which is something behind the scope of this PR -.